sympy/printing/c.py

```diff
@@ -409,7 +409,7 @@ def _print_For(self, expr):
         else:
             raise NotImplementedError("Only iterable currently supported is Range")
         body = self._print(expr.body)
-        return ('for ({target} = {start}; {target} < {stop}; {target} += '
+        return ('for (int {target} = {start}; {target} < {stop}; {target} += '
```

Comment:
I'm afraid that's not valid in C89 (and for example will fail to compile with MSVC).

Reply:
I initially created it as a local variable. Then, I rewrote the for loop as described here https://github.com/sympy/sympy/blob/master/sympy/printing/c.py#L714. But I didn't pay attention to the C version.

This means that `_get_loop_opening_ending` is not implemented for the C89 version.

I can add `_get_loop_opening_ending` to the C89 version and define two versions of `print_For`.

What do you think?

Reply:
I think that's fine, I guess we could compile as C++ if someone later wants to add support for MSVC?

Reply:
I would like to add this kind of tests into the external directory where the compilation process is performed. In that way, we should be able to make a choice if issues appear.

Reply:
That would be ideal. I suspect it will be a lot of work getting MSVC to run during the continuous integration tests. I know there are (at least were) some tests that test autowrap. For the printers and simpler "building blocks" in `sympy.codegen` I did add tests that at least were running on linux:
https://github.com/sympy/sympy/blob/6e4ee3392497a69b0ea453ca84ea53970f8a338b/sympy/codegen/tests/test_algorithms.py#L25

Reply:
So, the conclusion is that I let the `int` type into the loop and I will update the implementation during future compilation tests.

Is it good for you?

Reply:
I guess, but I'm not quite sure what you mean with "let the `int` type into the loop". For the C89 printer we cannot declare it there, but for C99 and later it would work.

Reply:
Now, the for loops work for both C89 and C99.
I can make C99 working exactly as before but then I have to modify the `CCodeGen` a little.